samples: mesh: nrf52: Corrected & improved transition implementation

vikrant8052 opened the pull request:
1. User can set transition time as Zero & delay as non-zero value.
    In that case, we have to depend upon timer.
2. Added some Kconfig parameters in prj.conf
3. Corrected & improved transition implementation so that
no_transition_work thread will not get triggered twice at the
end of transition.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: considered delay even in case of inst. transition

User can set transition time as Zero & delay as non-zero value.
In that case, we have to depend upon timer.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: added some Kconfig

Added some Kconfig which could help to improve Mesh
performance or Debugging.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: improved comment

Improved comment content.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
In hurry I forgot to test the result for condition where user could set transition time(tt) = 0 & Delay = non-zero value. In such transition, we could not consider it as fully instantaneous so we have to depend upon timer for proper working.

vikrant8051 commented:
Please don't merge as of now.
I've to add one more commit.

vikrant8052 committed:
samples: mesh: nrf52: corrected & improved transition implementation

Corrected & improved transition implementation so that
no_transition_work thread will not get triggered twice at the
end of transition.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
I've submitted last commit from my side.

vikrant8051 commented:
@trond-snekvik Please review.

trond-snekvik commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/transition.c:
```diff
@@ -257,44 +267,51 @@ static void light_lightness_actual_work_handler(struct k_work *work)
+
+		if (ctl->transition->counter == 0U) {
+			update_light_state();
+			k_timer_stop(&ctl->transition->timer);
```
Should this do `ctl->light->current = ctl->light-target` as well?

trond-snekvik replied:
Right, I see

vikrant8051 commented:
@jhedberg Please merge.
